Cannot find module 'gatsby-cli/lib/reporter'

After updating from gatsby 2.24.36  to version 2.24.40 getting error on npm start:
Error in "/Users/Austin/Projects/schwayweb/node_modules/gatsby-plugin-page-creator/gatsby-node.js": Cannot find module 'gatsby-cli/lib/reporter'

entire output here: https://gist.github.com/theskillwithin/2482854d8aa23e91885799925516c833


```

  System:
    OS: macOS 10.15.6
    CPU: (12) x64 Intel(R) Xeon(R) CPU E5-1650 v2 @ 3.50GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.14.1 - ~/.nvm/versions/node/v12.14.1/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.6 - ~/.nvm/versions/node/v12.14.1/bin/npm
  Languages:
    Python: 2.7.17 - /usr/local/bin/python
  Browsers:
    Chrome: 84.0.4147.105
    Firefox: 79.0
    Safari: 13.1.2
  npmPackages:
    gatsby: ^2.24.40 => 2.24.40 
    gatsby-background-image: ^1.1.1 => 1.1.1 
    gatsby-image: ^2.4.15 => 2.4.15 
    gatsby-plugin-emotion: ^4.3.10 => 4.3.10 
    gatsby-plugin-google-analytics: ^2.3.13 => 2.3.13 
    gatsby-plugin-layout: ^1.3.10 => 1.3.10 
    gatsby-plugin-manifest: ^2.4.22 => 2.4.22 
    gatsby-plugin-offline: ^3.2.23 => 3.2.23 
    gatsby-plugin-react-helmet: ^3.3.10 => 3.3.10 
    gatsby-plugin-root-import: ^2.0.5 => 2.0.5 
    gatsby-plugin-sharp: ^2.6.26 => 2.6.26 
    gatsby-source-contentful: ^2.3.36 => 2.3.36 
    gatsby-source-filesystem: ^2.3.24 => 2.3.24 
    gatsby-source-shopify: ^3.2.26 => 3.2.26 
    gatsby-transformer-remark: ^2.8.28 => 2.8.28 
    gatsby-transformer-sharp: ^2.5.12 => 2.5.12 
```

> OK - weird solution, but it might help others: removed package-lock.json and node_modules, then ran `yarn install`
> Checked that into github and Gatsby Cloud built the site.

When I run `gatsby develop` the followin error shows up:

```
[blyatmobilebr@blyatmobilebr-pc coffee-shop]$ gatsby develop
Debugger listening on ws://127.0.0.1:9229/64580fc3-4d58-4652-9ae6-6648f2debd34
For help, see: https://nodejs.org/en/docs/inspector
success open and validate gatsby-configs - 0.125s

 ERROR 

Error in "/home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby-plugin-page-creator/gatsby-node.js": Cannot find module 'gatsby-cli/lib/reporter'
Require stack:
- /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby-plugin-page-creator/is-valid-collection-path-implementation.js
- /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby-plugin-page-creator/create-pages-from-collection-builder.js
- /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby-plugin-page-creator/create-page-wrapper.js
- /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby-plugin-page-creator/gatsby-node.js
- /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/bootstrap/resolve-module-exports.js
- /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/bootstrap/load-plugins/validate.js
- /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/bootstrap/load-plugins/load.js
- /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/bootstrap/load-plugins/index.js
- /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/services/initialize.js
- /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/services/index.js
- /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/state-machines/develop/services.js
- /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/state-machines/develop/index.js
- /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/commands/develop-process.js
- /home/blyatmobilebr/Área de trabalho/coffee-shop/.cache/tmp-12398-m3KLIrnjoKyl



  Error: Cannot find module 'gatsby-cli/lib/reporter'
  Require stack:
  - /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby-plugin-page-creator/is-valid-col  lection-path-implementation.js
  - /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby-plugin-page-creator/create-pages  -from-collection-builder.js
  - /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby-plugin-page-creator/create-page-  wrapper.js
  - /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby-plugin-page-creator/gatsby-node.  js
  - /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/bootstrap/resolve-module-ex  ports.js
  - /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/bootstrap/load-plugins/vali  date.js
  - /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/bootstrap/load-plugins/load  .js
  - /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/bootstrap/load-plugins/inde  x.js
  - /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/services/initialize.js
  - /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/services/index.js
  - /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/state-machines/develop/serv  ices.js
  - /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/state-machines/develop/inde  x.js
  - /home/blyatmobilebr/Área de trabalho/coffee-shop/node_modules/gatsby/dist/commands/develop-process.js  - /home/blyatmobilebr/Área de trabalho/coffee-shop/.cache/tmp-12398-m3KLIrnjoKyl
  
  - loader.js:1030 Function.Module._resolveFilename
    internal/modules/cjs/loader.js:1030:15
  
  - loader.js:899 Function.Module._load
    internal/modules/cjs/loader.js:899:27
  
  - loader.js:1090 Module.require
    internal/modules/cjs/loader.js:1090:19
  
  - v8-compile-cache.js:159 require
    [coffee-shop]/[v8-compile-cache]/v8-compile-cache.js:159:20
  
  - is-valid-collection-path-implementation.js:10 Object.<anonymous>
    [coffee-shop]/[gatsby-plugin-page-creator]/is-valid-collection-path-implementation.js:10:40
  
  - v8-compile-cache.js:178 Module._compile
    [coffee-shop]/[v8-compile-cache]/v8-compile-cache.js:178:30
  
  - loader.js:1221 Object.Module._extensions..js
    internal/modules/cjs/loader.js:1221:10
  
  - loader.js:1050 Module.load
    internal/modules/cjs/loader.js:1050:32
  
  - loader.js:938 Function.Module._load
    internal/modules/cjs/loader.js:938:14
  
  - loader.js:1090 Module.require
    internal/modules/cjs/loader.js:1090:19
  
  - v8-compile-cache.js:159 require
    [coffee-shop]/[v8-compile-cache]/v8-compile-cache.js:159:20
  
  - create-pages-from-collection-builder.js:28 Object.<anonymous>
    [coffee-shop]/[gatsby-plugin-page-creator]/create-pages-from-collection-builder.js:28:44
  
  - v8-compile-cache.js:178 Module._compile
    [coffee-shop]/[v8-compile-cache]/v8-compile-cache.js:178:30
  
  - loader.js:1221 Object.Module._extensions..js
    internal/modules/cjs/loader.js:1221:10
  
  - loader.js:1050 Module.load
    internal/modules/cjs/loader.js:1050:32
  
  - loader.js:938 Function.Module._load
    internal/modules/cjs/loader.js:938:14
  

not finished load plugins - 5.449s
```

--- a/packages/gatsby-plugin-page-creator/src/__tests__/is-valid-collection-path-implementation.ts
+++ b/packages/gatsby-plugin-page-creator/src/__tests__/is-valid-collection-path-implementation.ts
@@ -15,7 +15,7 @@ describe(`isValidCollectionPathImplementation`, () => {
     `%o passes`,
     path => {
       expect(() =>
-        isValidCollectionPathImplementation(compatiblePath(path))
+        isValidCollectionPathImplementation(compatiblePath(path), reporter)
       ).not.toThrow()
     }
   )
@@ -29,7 +29,7 @@ describe(`isValidCollectionPathImplementation`, () => {
   ])(`%o throws as expected`, path => {
     const part = path.split(`/`)[2]
 
-    isValidCollectionPathImplementation(compatiblePath(path))
+    isValidCollectionPathImplementation(compatiblePath(path), reporter)
     expect(reporter.panicOnBuild)
       .toBeCalledWith(`Collection page builder encountered an error parsing the filepath. To use collection paths the schema to follow is {Model.field}. The problematic part is: ${part}.
 filePath: ${compatiblePath(path)}`)

--- a/packages/gatsby-plugin-page-creator/src/create-page-wrapper.ts
+++ b/packages/gatsby-plugin-page-creator/src/create-page-wrapper.ts
@@ -4,6 +4,7 @@ import { createClientOnlyPage } from "./create-client-only-page"
 import { createPagesFromCollectionBuilder } from "./create-pages-from-collection-builder"
 import systemPath from "path"
 import { trackFeatureIsUsed } from "gatsby-telemetry"
+import { Reporter } from "gatsby"
 
 function pathIsCollectionBuilder(path: string): boolean {
   return path.includes(`{`)
@@ -18,7 +19,8 @@ export function createPage(
   pagesDirectory: string,
   actions: Actions,
   ignore: string[],
-  graphql: CreatePagesArgs["graphql"]
+  graphql: CreatePagesArgs["graphql"],
+  reporter: Reporter
 ): void {
   // Filter out special components that shouldn't be made into
   // pages.
@@ -41,7 +43,13 @@ export function createPage(
         `PageCreator: Found a collection route, but the proper env was not set to enable this experimental feature. Please run again with \`GATSBY_EXPERIMENTAL_ROUTING_APIS=1\` to enable.`
       )
     }
-    createPagesFromCollectionBuilder(filePath, absolutePath, actions, graphql)
+    createPagesFromCollectionBuilder(
+      filePath,
+      absolutePath,
+      actions,
+      graphql,
+      reporter
+    )
     return
   }
 

--- a/packages/gatsby-plugin-page-creator/src/create-pages-from-collection-builder.ts
+++ b/packages/gatsby-plugin-page-creator/src/create-pages-from-collection-builder.ts
@@ -8,18 +8,24 @@ import { derivePath } from "./derive-path"
 import { watchCollectionBuilder } from "./watch-collection-builder"
 import { collectionExtractQueryString } from "./collection-extract-query-string"
 import { isValidCollectionPathImplementation } from "./is-valid-collection-path-implementation"
-import reporter from "gatsby-cli/lib/reporter"
 
 // TODO: Do we need the ignore argument?
 export async function createPagesFromCollectionBuilder(
   filePath: string,
   absolutePath: string,
   actions: Actions,
-  graphql: CreatePagesArgs["graphql"]
+  graphql: CreatePagesArgs["graphql"],
+  reporter
 ): Promise<void> {
-  if (isValidCollectionPathImplementation(absolutePath) === false) {
+  if (isValidCollectionPathImplementation(absolutePath, reporter) === false) {
     watchCollectionBuilder(absolutePath, ``, [], actions, () =>
-      createPagesFromCollectionBuilder(filePath, absolutePath, actions, graphql)
+      createPagesFromCollectionBuilder(
+        filePath,
+        absolutePath,
+        actions,
+        graphql,
+        reporter
+      )
     )
     return
   }
@@ -30,7 +36,13 @@ export async function createPagesFromCollectionBuilder(
   // 1.a  If the query string is not findable, we can't move on. So we stop and watch
   if (queryString === null) {
     watchCollectionBuilder(absolutePath, ``, [], actions, () =>
-      createPagesFromCollectionBuilder(filePath, absolutePath, actions, graphql)
+      createPagesFromCollectionBuilder(
+        filePath,
+        absolutePath,
+        actions,
+        graphql,
+        reporter
+      )
     )
     return
   }
@@ -51,7 +63,13 @@ ${errors.map(error => error.message).join(`\n`)}`.trim()
     )
 
     watchCollectionBuilder(absolutePath, queryString, [], actions, () =>
-      createPagesFromCollectionBuilder(filePath, absolutePath, actions, graphql)
+      createPagesFromCollectionBuilder(
+        filePath,
+        absolutePath,
+        actions,
+        graphql,
+        reporter
+      )
     )
 
     return
@@ -97,6 +115,12 @@ ${errors.map(error => error.message).join(`\n`)}`.trim()
   })
 
   watchCollectionBuilder(absolutePath, queryString, paths, actions, () =>
-    createPagesFromCollectionBuilder(filePath, absolutePath, actions, graphql)
+    createPagesFromCollectionBuilder(
+      filePath,
+      absolutePath,
+      actions,
+      graphql,
+      reporter
+    )
   )
 }

--- a/packages/gatsby-plugin-page-creator/src/create-pages-from-collection-builder.ts
+++ b/packages/gatsby-plugin-page-creator/src/create-pages-from-collection-builder.ts
@@ -8,6 +8,7 @@ import { derivePath } from "./derive-path"
 import { watchCollectionBuilder } from "./watch-collection-builder"
 import { collectionExtractQueryString } from "./collection-extract-query-string"
 import { isValidCollectionPathImplementation } from "./is-valid-collection-path-implementation"
+import { Reporter } from "gatsby"
 
 // TODO: Do we need the ignore argument?
 export async function createPagesFromCollectionBuilder(
@@ -15,7 +16,7 @@ export async function createPagesFromCollectionBuilder(
   absolutePath: string,
   actions: Actions,
   graphql: CreatePagesArgs["graphql"],
-  reporter
+  reporter: Reporter
 ): Promise<void> {
   if (isValidCollectionPathImplementation(absolutePath, reporter) === false) {
     watchCollectionBuilder(absolutePath, ``, [], actions, () =>

--- a/packages/gatsby-plugin-page-creator/src/gatsby-node.ts
+++ b/packages/gatsby-plugin-page-creator/src/gatsby-node.ts
@@ -67,7 +67,7 @@ Please pick a path to an existing directory.`)
   // Get initial list of files.
   let files = await glob(pagesGlob, { cwd: pagesPath })
   files.forEach(file => {
-    createPage(file, pagesDirectory, actions, ignore, graphql)
+    createPage(file, pagesDirectory, actions, ignore, graphql, reporter)
   })
 
   watchDirectory(
@@ -75,7 +75,14 @@ Please pick a path to an existing directory.`)
     pagesGlob,
     addedPath => {
       if (!_.includes(files, addedPath)) {
-        createPage(addedPath, pagesDirectory, actions, ignore, graphql)
+        createPage(
+          addedPath,
+          pagesDirectory,
+          actions,
+          ignore,
+          graphql,
+          reporter
+        )
         files.push(addedPath)
       }
     },

--- a/packages/gatsby-plugin-page-creator/src/is-valid-collection-path-implementation.ts
+++ b/packages/gatsby-plugin-page-creator/src/is-valid-collection-path-implementation.ts
@@ -1,12 +1,15 @@
 import sysPath from "path"
-import reporter from "gatsby-cli/lib/reporter"
+import { Reporter } from "gatsby"
 
 // This file is a helper for consumers. It's going to log an error to them if they
 // in any way have an incorrect filepath setup for us to predictably use collection
 // querying.
 //
 // Without this, users will can get mystic errors.
-export function isValidCollectionPathImplementation(filePath: string): boolean {
+export function isValidCollectionPathImplementation(
+  filePath: string,
+  reporter: Reporter
+): boolean {
   const parts = filePath.split(sysPath.sep)
   let passing = true
 

